pkg/discovery/worker/compliance/affinity_processor.go

```diff
@@ -46,9 +48,11 @@ func (am *AffinityProcessor) processAffinityPerPod(pod *api.Pod, podsNodesMap ma
 	// Honor the nodeAffinity from the pod's nodeAffinity
 	nodeSelectorTerms := getAllNodeSelectors(affinity)
 	// Also honor the nodeAffinity from the PVs of the pod if the pod have the PV attached
-	pvNodeSelectorTerms := am.getAllPvAffinityTerms(pod)
-
-	nodeSelectorTerms = append(nodeSelectorTerms, pvNodeSelectorTerms...)
+	var pvNodeSelectorTerms []api.NodeSelectorTerm
+	if utilfeature.DefaultFeatureGate.Enabled(features.HonorAzLabelPvAffinity) {
```

Comment:
yes, `match` only works when `nodeSelector ` has a length greater than 0, otherwise `matchesPvNodeAffinity` will return true, not impacting other logic. 
```
	if len(pvNodeAffinitySelectorTerms) > 0 {
		nodeAffinityMatches = nodeMatchesNodeSelectorTerms(node, pvNodeAffinitySelectorTerms)
	}
```